lib/perl/Genome/InstrumentData/AlignmentResult.pm

```diff
@@ -1629,9 +1629,112 @@ sub verify_alignment_data {
 }
 
 sub alignment_bam_file_paths {
+    return glob(shift->output_dir . "/*.bam");
+}
+
+sub recreated_alignment_bam_file_paths {
+    my $self = shift;
+
+    my @bams = glob($self->output_dir . "/*.bam");
+    if (@bams) {
+        return @bams;
+    }
+
+    my $recreated_bam = File::Spec->join($self->output_dir, 'all_sequences.bam');
+    my $merged_bam    = $self->get_merged_bam_to_revivify_per_lane_bam;
```

Comment:
Yeah, we werent very happy with the method name `recreated_alignment_bam_file_paths` in general. I want the method name to be something that means 'I'll get you the file paths, whether they exist yet or not' as opposed to the method above that only returns them if they exist.

I agree that revivify is both cooler and perhaps more appropriate than recreate (arguably). Unsure if it's unnecessarily confusing / fancy. I agree the same word should be used everywhere and I'll fix that once we figure out a good name for this method in general.